zsl/interface/celery/worker.py

```diff
@@ -0,0 +1,101 @@
+"""
+:mod:`zsl.interface.celery.worker`
+----------------------------------
+
+Implementation of celery workers.
+
+.. moduleauthor:: Peter Morihladko
+"""
+from __future__ import unicode_literals
+
+import sys
+
+from celery import Celery, shared_task
+from injector import Module, singleton, provides
+
+from zsl.utils.injection_helper import inject
+from zsl.interface.task_queue import TaskQueueWorker
+from zsl.task.job_context import Job
+
+
+class CeleryTaskQueueWorkerBase(TaskQueueWorker):
+    """Base class for celery task queue worker.
+
+    It contains only the task execution part of worker.
+    """
+
+    def execute_celery_task(self, job_data):
+        # type: (dict) -> dict
+        """Creates job from task and executes the job.
+
+        :param job_data: job's data
+        :return: job results
+        :rtype: dict
+        """
+        job = Job(job_data)
+
+        return self.execute_job(job)
+
+
+class CeleryTaskQueueOutsideWorker(CeleryTaskQueueWorkerBase):
+    """Celery worker used only for task execution.
+
+    This can be used when the worker is manage with some other application,
```

Comment:
When you run worker using other "runners", e. g.  `celery worker -A app.celery` or ` celery multi start 1 -A app.celery` there are two "systems" in play. Celery expects an `app` objects with associated celery tasks and will run as worker.